main.nf

```diff
@@ -933,147 +1165,167 @@ process busco_download_db {
-            --out ${assembly} \
-            >${assembly}_busco_log.txt
-        cp run_${assembly}/short_summary_${assembly}.txt short_summary_${assembly}.txt
-
-        for f in run_${assembly}/single_copy_busco_sequences/*faa; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*faa >${assembly}_buscos.faa || touch ${assembly}_buscos.faa
-            break
-        done
-        for f in run_${assembly}/single_copy_busco_sequences/*fna; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*fna >${assembly}_buscos.fna || touch ${assembly}_buscos.fna
-            break
-        done
-        """
-    } else {
-        """
-        run_BUSCO.py \
-            --in ${assembly} \
-            --lineage_path $db_name \
-            --cpu "${task.cpus}" \
-            --blast_single_core \
-            --mode genome \
-            --out ${assembly} \
-            >${assembly}_busco_log.txt
-        cp run_${assembly}/short_summary_${assembly}.txt short_summary_${assembly}.txt
-
-        for f in run_${assembly}/single_copy_busco_sequences/*faa; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*faa >${assembly}_buscos.faa || touch ${assembly}_buscos.faa
-            break
-        done
-        for f in run_${assembly}/single_copy_busco_sequences/*fna; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*fna >${assembly}_buscos.fna || touch ${assembly}_buscos.fna
-            break
-        done
-        """
-    }
+    # place db in extra folder to ensure BUSCO recognizes it as path (instead of downloading it)
+    mkdir dataset
+    mv ${db} dataset/
+
+    busco --lineage_dataset dataset/${db} \
+        --mode genome \
+        --in ${bin} \
+        --config \${config_file} \
+        --cpu "${task.cpus}" \
+        --out "BUSCO" > ${bin}_busco.log
+
+    # get used db name
+    # (set nullgob: if pattern matches no files, expand to a null string rather than to itself)
+    shopt -s nullglob
+    summaries=(BUSCO/short_summary.specific.*.BUSCO.txt)
+    if [ \${#summaries[@]} -ne 1 ]; then
+        echo "ERROR: none or multiple 'BUSCO/short_summary.specific.*.BUSCO.txt' files found. Expected one."
+        exit 1
+    fi
+    [[ \$summaries =~ BUSCO/short_summary.specific.(.*).BUSCO.txt ]];
+    db_name="\${BASH_REMATCH[1]}"
+    echo "Used database: \${db_name}"
+
+    cp BUSCO/short_summary.specific.\${db_name}.BUSCO.txt short_summary.specific.\${db_name}.${bin}.txt
+
+    for f in BUSCO/run_\${db_name}/busco_sequences/single_copy_busco_sequences/*faa; do
+        cat BUSCO/run_\${db_name}/busco_sequences/single_copy_busco_sequences/*faa | gzip >${bin}_buscos.faa.gz
+        break
+    done
+    for f in BUSCO/run_\${db_name}/busco_sequences/single_copy_busco_sequences/*fna; do
+        cat BUSCO/run_\${db_name}/busco_sequences/single_copy_busco_sequences/*fna | gzip >${bin}_buscos.fna.gz
+        break
```

Comment:
This is so long that it might make sense to add it in a separate bash file in `bin`. But could also be left for the next release